code/modules/modular_computers/file_system/programs/medical/dna_analyzer.dm

```diff
@@ -0,0 +1,193 @@
+#define SCREEN_SETTINGS		"settings"
+#define SCREEN_DNALIST		"dnalist"
+#define SCREEN_WORKING		"working"
+#define SCREEN_RESEARCH		"research"
+#define SCREEN_LOCKED		"locked"
+
+/datum/computer_file/program/dna_analyzer
+	filename = "dnaAnalyzer"
+	filedesc = "DNA Analyzing program"
+	extended_desc = "This program allows analyze of deoxyribonucleic acid. This program can not be run without DNA processor installed."
+	required_access = access_genetics
+	program_icon_state = "dna"
+	program_key_state = "med_key"
+	program_menu_icon = "link"
+	usage_flags = PROGRAM_CONSOLE
+	size = 1
+	requires_ntnet = 0
+	available_on_ntnet = 1
+	nanomodule_path = /datum/nano_module/dna_analyzer
+
+/datum/computer_file/program/dna_analyzer/run_program()
+	. = ..()
+
+	if(NM)
+		var/datum/nano_module/dna_analyzer/NME = NM
+		NME.linked_dna_processor = computer.dna_processor
+
+/datum/computer_file/program/dna_analyzer/process_tick()
+	..()
+	var/datum/nano_module/dna_analyzer/DNM = NM
+	if(istype(DNM))
+		DNM.emagged = computer_emagged
```

Comment:
Why do that? I don't get the `nano_module` thing at all, and I don't get why a module can't check that at program if necessary, as opposed to the program shoving that value there every tick.